src/erf.jl

```diff
@@ -362,6 +362,46 @@ function _erfinv(x::Float32)
     end
 end
 
+function _erfinv(x::Float16)
+    a = abs(x)
+    if a >= Float16(1)
+        if x == Float16(1)
+            return Inf16
+        elseif x == -Float16(1)
+            return -Inf16
+        end
+        throw(DomainError(a, "`abs(x)` cannot be greater than 1."))
+    end
+
+    # Perform calculations with `Float32`
+    x32 = Float32(x)
+    a32 = Float32(a)
+    if a32 <= 0.75f0 # Table 7 in Blair et al.
+        t = x32^2 - 0.5625f0
+        y = x32 * @horner(t, -0.10976_672f1,
+                              0.53062_1f0) /
+                  @horner(t, -0.10123_953f1,
+                              0.1f1)
+    elseif a32 <= 0.9375f0 # Table 26 in Blair et al.
+        t = x32^2 - 0.87890625f0
+        y = x32 * @horner(t, 0.10178_950f1,
+                            -0.32827_601f1) /
+                  @horner(t, 0.72455_99f0,
+                            -0.33871_553f1,
+                             0.1f1)
+    else # Table 47 in Blair et al.
+        t = inv(sqrt(-log1p(-a32)))
+        y = @horner(t, 0.98650_088f0,
+                       0.92601_777f0) /
+            (copysign(t, x32) *
+             @horner(t, 0.98424_719f0,
+                        0.10074_7432f0,
+                        0.1f0))
```

Comment:
```suggestion
t = sqrt(-log(1-a32))
y = copysign(@horner(t, -0.429159f0, 1.04868f0), x32)
```
This can be a lot simpler because of the reduced accuracy requirement (it turns out that this branch only covers 128 values). Note that the use of `log(1-x)` rather than `log1p` is intentional since it will be slightly faster and `1-x` is exact.

Reply:
Done, thanks!